charts/common/templates/deployment.yaml

```diff
@@ -302,22 +303,47 @@ spec:
             {{- toYaml .Values.securityContext | nindent 12 }}
           image: "{{ .Values.cloudsqlProxy.image.repository }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if ( eq $sqlProxyVersion "v2" ) }}
+          ports:
+            {{- toYaml .Values.cloudsqlProxy.ports | nindent 12 }}
+          livenessProbe:
+            {{- toYaml .Values.cloudsqlProxy.livenessProbe | nindent 12 }}
+          readinessProbe:
+            {{- toYaml .Values.cloudsqlProxy.readinessProbe | nindent 12 }}
+          startupProbe:
+            {{- toYaml .Values.cloudsqlProxy.startupProbe | nindent 12 }}
+          lifecycle:
+            {{- toYaml .Values.cloudsqlProxy.lifecycle | nindent 12 }}
+          {{- end }}
           resources:
             {{- toYaml .Values.cloudsqlProxy.resources | nindent 12 }}
-          {{- if .Values.cloudsqlProxy.command }}
+          {{- if and ( eq $sqlProxyVersion "v1" ) ( .Values.cloudsqlProxy.instanceConnectionName ) ( not .Values.cloudsqlProxy.command ) }}
```

Comment:
@ndresselhaus added this logic to prevent this problem
- someone’s got an existing setup without specifying the repository because the default is
- we do a mass upgrade on charts
- the command/args break because 1.0 -> 2.0 is different

This ensures we only use args with v2 and command with v1 with the option to overwrite